interaction-manager/src/lib_components/spatial-navigation.js

```diff
@@ -246,30 +248,21 @@ class SpatialNavigation {
      * Registers actions and adds them to the keyboard and gamepad objects
     */
     registerKeyActions() {
-        const hasCustomKeys = Object.entries(this.customKeys).length !== 0;
-
         directions.forEach((direction) => {
             const callback = () => {
                 this.moveFocus(direction);
             };
             actions.register(`move-focus-${direction}`, callback);
 
-            let keys = [[`arrow_${direction}`]];
-
-            if (hasCustomKeys && this.customKeys[direction]) {
-                if (this.changeDefault) {
-                    keys = [[this.customKeys[direction]]];
-                } else {
-                    keys.push([this.customKeys[direction]]);
-                }
-            }
+            const keys = this.activeKeys[direction].map(key => [key]);
 
-            for (let i = 0; i < keys.length; i++) {
+            for (const key of keys) {
                 keyboard.on({
-                    keys: keys[i],
+                    keys: key,
                     callback: `move-focus-${direction}`,
                     type: 'press',
                 });
+                this.registeredKeys.add(key[0]);
             }

```

Comment:
```
const keys = this.activeKeys[direction];

for (const key of keys) {
   keyboard.on({
       keys: [key],
       callback: `move-focus-${direction}`,
       type: 'press',
   });
   
  this.registeredKeys.add(key[0]);
}
```

Isn't this the same but simpler? And also you omit calling the `.map` method.

Reply:
Fixed